src/app-host/app-host.js

```diff
@@ -5,93 +5,187 @@ var Messages = require('messages');
+            }
+        }
+    });
+}
+
 function setCordova(originalCordova) {
+    var channel,
+        platform,
+        platformBoostrap;
+
     if (cordova) {
         return;
     }
 
     cordova = originalCordova;
 
-    oldExec = cordova.require('cordova/exec');
     cordova.define.remove('cordova/exec');
     cordova.define('cordova/exec', function (require, exports, module) {
         module.exports = exec;
     });
 
-    // android platform has its own specific initialization
-    // so to emulate it, we need to fake init function
-    if (cordova.platformId === 'android') {
-        exec.init = function () {
-            cordova.require('cordova/channel').onNativeReady.fire();
-        };
-    }
-
-    // windows phone platform fires 'deviceready' event from native component
-    // so to emulate it we fire it in the bootstrap function (similar to ios)
-    if (cordova.platformId === 'windowsphone') {
-        cordova.require('cordova/platform').bootstrap = function () {
-            cordova.require('cordova/channel').onNativeReady.fire();
-        };
-    }
+    platform = cordova.require('cordova/platform');
+    platformBootstrap = platform.bootstrap;
+    platform.bootstrap = function () {
+    };
 
     // default Windows bootstrap function tries to load WinJS which is not
     // available and not required in simulation mode so we override bootstrap
     if (cordova.platformId === 'windows') {
-        cordova.require('cordova/platform').bootstrap = function () {
+        platformBootstrap = function () {
             cordova.require('cordova/modulemapper')
                 .clobbers('cordova/exec/proxy', 'cordova.commandProxy');
 
-            cordova.require('cordova/channel').onNativeReady.fire();
         };
     }
-}
 
-function getCordova() {
-    return cordova;
-}
+    channel = cordova.require('cordova/channel');
+
+    // define our own channel to delay the initialization until sim-host tells
+    // us everything's ready (fired in 'start' event handler).
+    channel.createSticky('onCordovaSimulateReady');
+    channel.waitForInitialization('onCordovaSimulateReady');
+
+    // firing of onNativeReady is delayed until SIM_HOST tells us it's ready
+    socket.once('init', function () {
+        // sim-host is ready, register exec handlers, fire onNativeReady and send
+        // the list of plugins
+        socket.on('exec-success', function (data) {
+            console.log('exec-success:');
+            console.log(data);
+            var execCacheInfo = execCache[data.index];
+            if (execCacheInfo.success) {
+                execCacheInfo.success(data.result);
+            }
+        });
+
+        socket.on('exec-failure', function (data) {
+            console.log('exec-failure:');
+            console.log(data);
+            var execCacheInfo = execCache[data.index];
+            if (execCacheInfo.fail) {
+                execCacheInfo.fail(data.error);
+            }
```

Comment:
I think it's a good idea to check `execCacheInfo` is actually there. Type check seems to be excessive for me: it would be a plugin simulation bug. We need to simplify a task of adding simulation support for a plugin author, it would be helpful there. I'm thinking of whether it needs to be done now or later (as part of the "developing simulation support" feature later, for example).

Anyways, I agree... I think this should be the condition for the happy scenario:
```javascript
if (execCacheInfo && typeof execCacheInfo.fail === 'function') {
```